subsys/testsuite/cmock/CMakeLists.txt

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2022 Swedish Embedded Consulting Group AB
+# Author: Martin Schröder <martin.schroder@swedishembedded.com>
+# SPDX-License-Identifier: Apache-2.0
+# Consulting: https://swedishembedded.com/go
+# Training: https://swedishembedded.com/tag/training
+
+zephyr_library()
+
+# set the directory where cmock has been checked out by west
+set_property(GLOBAL PROPERTY CMOCK_DIR ${ZEPHYR_BASE}/../modules/test/cmock)
+# need to also import it into current file so it can be used
+get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+
+# locate cmock config file
+set(CMOCK_CONFIG_FILE
+    ${CMAKE_CURRENT_LIST_DIR}/cmock_cfg.yaml
+    CACHE STRING "")
+
+# check if we have ruby executable
+find_program(RUBY_PATH ruby)
+if(${RUBY_PATH} STREQUAL RUBY_PATH-NOTFOUND)
```

Comment:
If we're adding an example (in `tests/cmock/example`) I believe we should have some more or less automated way to get all the required software to run cmock, which this PR is currently lacking.

Reply:
> Note that I'm building this on the Zephyr docker image. Could you describe in more detail how to prepare the environment to be able to build this sample?

You will need these packages: https://github.com/swedishembedded/develop/blob/d6a16063c30d465caf35adc5bc165ad541e80197/Dockerfile.build#L52

And also probably clang pip package: https://github.com/swedishembedded/develop/blob/d6a16063c30d465caf35adc5bc165ad541e80197/Dockerfile.build#L22

Basically the script uses libclang to extract function names from headers.

Reply:
What would be a standard procedure for making CI pass here? Should I raise a PR at the docker image repo first with the changes or should I simply add missing packages to ci scripts here and then somebody else will add them to the docker image?